src/base/communication/communicationBase_mpi.cpp

```diff
@@ -677,7 +677,7 @@ void CommunicationBase::initIOBinary(std::string fileName, size_t filesize, size
     if (mode == WRITE) mpi_mode = MPI_MODE_WRONLY;
     if (mode == READWRITE) mpi_mode = MPI_MODE_RDWR;
 
-    if (MPI_File_open(getCart_comm(), const_cast<char *>(fileName.c_str()),
+    if (MPI_File_open(MPI_COMM_WORLD, const_cast<char *>(fileName.c_str()),
```

Comment:
I mentioned this in an e-mail to Rasmus, so I'll copy my response.

"Without this change, I get stack smashing errors on the GPU machine at
MSU.  I traced it with gdb to this line.  I did confirm today that the
crash still happens with my latest commit, making the change back to
getCart_comm().  The crash occurs regardless of how many mpi processes
I run. 
"

So in summary I don't know why it fails for getCart_comm(), but since it is just opening a file perhaps it is ok to use the global MPI communicator.

Reply:
That sounds like a special problem with the MSU machine. Could you please check whether this is also an issue on another machine (e.g. Bielefeld Cluster) ?

Reply:
Sure.  Does the Bielefeld cluster have a GPU node?  I do have an account on the Bielefeld cluster, but I don't currently have access to any compute resources there, I think.  I have just been using my account there to back up data to tape.

Reply:
Yes, if you have an account you can use our GPUs. It looks like you have a Redmine account, which means you can learn how to use our GPU cluster here: https://rmp.physik.uni-bielefeld.de/projects/cluster

On our Volta machine, which is equipped with V100 GPUs, you can get a 4-GPU interactive node with
`run --partition=volta_devel --nodes=1 --ntasks=4 --gres-flags=enforce-binding --gres=gpu:4 --time=00:30:00 --gpus=4 --pty $SHELL`